src/components/Pagination/Pagination.tsx

```diff
@@ -2,0 +1,79 @@
+import classNames from 'classnames';
+import React from 'react';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const totalPages = Math.ceil(total / perPage);
+
+  const handlePageChange = (page: number) => {
+    if (page < 1 || page > totalPages || page === currentPage) {
+      return;
+    }
+
+    onPageChange(page);
+  };
+
+  return (
+    <ul className="pagination">
+      <li
+        className={classNames('page-item', {
+          disabled: currentPage === 1,
+        })}
+      >
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          onClick={() => handlePageChange(currentPage - 1)}
+          aria-disabled={currentPage === 1}
+        >
+          «
+        </a>
+      </li>
+
+      {Array.from({ length: totalPages }, (_, index) => (
+        <li
+          className={classNames('page-item', {
+            active: currentPage === index + 1,
+          })}
+          key={index}
+        >
+          <a
+            data-cy="pageLink"
+            className="page-link"
+            href={`#${index + 1}`}
```

Comment:
Using `href` attributes for pagination links can cause issues with accessibility and page behavior. Consider using `button` elements for pagination controls to avoid unintended navigation.